mail/common/src/main/java/com/fsck/k9/mail/ssl/LocalKeyStore.java

```diff
@@ -11,63 +11,37 @@
 import java.security.cert.CertificateException;
 import java.security.cert.X509Certificate;
 
+import android.content.Context;
+
 import org.apache.commons.io.IOUtils;
 import timber.log.Timber;
 
 
 public class LocalKeyStore {
     private static final int KEY_STORE_FILE_VERSION = 1;
 
-    private static String sKeyStoreLocation;
-
-    public static void setKeyStoreLocation(String directory) {
-        sKeyStoreLocation = directory;
-    }
-
-    private static class LocalKeyStoreHolder {
-        static final LocalKeyStore INSTANCE = new LocalKeyStore();
-    }
-
-    public static LocalKeyStore getInstance() {
-        return LocalKeyStoreHolder.INSTANCE;
+    public static LocalKeyStore createInstance(Context context) {
+        String keyStoreLocation = context.getDir("KeyStore", Context.MODE_PRIVATE).toString();
```

Comment:
This adds a dependency to an Android class. Ideally we want to push these Android dependencies as far "up" as possible.

Reply:
I considered this but couldn't find an obvious solution that doesn't add a lot of code. Do you have a suggestion on how to do that here? Introduce a new K9Context interface and specialize it for methods like this?